### DISCUSSION TITLE
.fda error

### DISCUSSION THREAD

**Discussion by antoniohupa** [marksgraham/OCT-Converter#97] [General]
Hi Mark, hope you're doing well.

Recently I've installed the package with the newest updates. When I tried to execute the demo example for .fda files it throws an error (I attach a file to see it). On the other hand, this errors does not replicate with .fds files. Could you give me some orientations about?

Thanks a lot?
A
[fda_error.pdf](https://github.com/marksgraham/OCT-Converter/files/11039304/fda_error.pdf)

**Comment by Oli4** [comment]
Hi Antonia, 

have you tried using `read_oct_volume_2()` function, maybe this is a quick fix for you problem otherwise @marksgraham has to take over. 

Best Oliver

**Comment by marksgraham** [comment]
Hi @antoniohupa 

Is this using your own data, or the file in the demo? And do you think this error didn't occur in a previous version?

It looks like it isn't finding any slices in your data. Could you run `print(oct_volume)` just after you read it?

If you're able to share this scan I could take a look too.

**Comment by antoniohupa** [comment]
Hi Mark

It happens with both, your and my own data. That did not happen with
previous versions of the library.
When I print(oct_volume) it returns the following:

<oct_converter.image_types.oct.OCTVolumeWithMetaData object at
0x0000022F5F8BAF40>



Regards

A

**Comment by shojaeinia** [comment]
Hi Mark, hope you're doing well.

Recently I've installed the package with the newest updates. When I tried to execute the demo example for .fda files it throws an error 👍 
TypeError                                 Traceback (most recent call last)
[<ipython-input-7-2fb02ad99c96>](https://localhost:8080/#) in <cell line: 8>()
      6 
      7 oct_volume = (
----> 8     fda.read_oct_volume()
      9 )  # returns an OCT volume with additional metadata if available
     10 oct_volume.peek(show_contours=True)  # plots a montage of the volume

2 frames
[/usr/local/lib/python3.10/dist-packages/oct_converter/readers/fda.py](https://localhost:8080/#) in read_any_info_and_make_dict(self, chunk_name)
    347         with open(self.filepath, "rb") as f:
    348             chunk_location, chunk_size = self.chunk_dict[chunk_name]
--> 349             f.seek(chunk_location)  # Set the chunk’s current position.
    350             raw = f.read()
    351             header_name = f"{chunk_name.decode().split('@')[-1].lower()}_header"

TypeError: 'list' object cannot be interpreted as an integer

**Reply by marksgraham** [reply]
Hi,

Could you check if this works for you with the example file from the Biobank, download link in the example script: https://github.com/marksgraham/OCT-Converter/blob/main/examples/demo_fda_extraction.py
